lib/expand/join.go

```diff
@@ -436,16 +452,16 @@ func (p *Peer) submit(req *installpb.ExecuteRequest) bool {
 }
 
 // execute executes either the complete operation or a single phase specified with req
-func (p *Peer) execute(req *installpb.ExecuteRequest) (err error) {
+func (p *Peer) execute(req *installpb.ExecuteRequest) (dispatcher.Status, error) {
 	p.WithField("req", req).Info("Execute.")
 	opCtx, err := p.operationContext()
 	if err != nil {
-		return trace.Wrap(err)
+		return dispatcher.StatusUnknown, trace.Wrap(err)
 	}
 	if req.Phase != nil {
-		return trace.Wrap(p.executePhase(p.ctx, *opCtx, *req.Phase, p.dispatcher))
+		return p.executePhase(p.ctx, *opCtx, *req.Phase, p.dispatcher)
```

Comment:
I wanted to ask you about this before: how come this logic of executing phases leaked into Peer from FSM engine? It is weird to me that now Peer also has "execute", "executePhase" and "executeConcurrently" methods.

Reply:
No logic has leaked but there might be some naming clashes.